File: data/compilers_comparison/src/examples/second_example/src/func33.c
#include "second_example.h" 
array_t* func33(array_t_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array149;
   if (pCounter > 0) {
      array149 = vars->data[--pCounter];
      array149->refC++;
      DEBUG_COPY(array149->id);
   } else {
      array149 = (array_t*)malloc(sizeof(array_t));
      array149->size = 335;
      array149->refC = 1;
      array149->id = 149;
      array149->data = (unsigned int*)malloc(array149->size*sizeof(unsigned int));
      memset(array149->data, 0, array149->size*sizeof(unsigned int));
      DEBUG_NEW(array149->id);
   }
   array_t_param params0;
   params0.size = 1;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   params0.data[0] = array149;
   array_t* array150 = func36(&params0, rng(), loopsFactor);
   DEBUG_RETURN(array150->id);
   free(params0.data);
   unsigned int loop44 = 0;
   unsigned int loopLimit44 = (rand()%loopsFactor)/3 + 1;
   for(; loop44 < loopLimit44; loop44++) {
      for (int i = 0; i < array149->size; i++) {
         array149->data[i]++;
      }
   }
   for (int i = 0; i < array150->size; i++) {
      if (array150->data[i] == 49) { 
         return array150;
      }
   }
   array149->refC--;
   if(array149->refC == 0) {
      free(array149->data);
      free(array149);
      DEBUG_FREE(array149->id);
   }
   return array150;
}

File: data/multilang_comparison/program_nim/src/func33.nim
proc func33*(vars: ptr ArrayTParam, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   var array181: ptr ArrayT
   if pCounter > 0:
      dec pCounter
      array181 = vars.data[pCounter]
      inc array181.refC
      debugCopy(array181.id)
   else:
      array181 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
      array181.size = 949
      array181.refC = 1
      array181.id = 181
      array181.data = cast[ptr UncheckedArray[uint32]](alloc(array181.size * sizeof(uint32)))
      zeroMem(array181.data, array181.size * sizeof(uint32))
      debugNew(array181.id)
   var params0: ArrayTParam
   params0.size = 1
   params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
   params0.data[0] = array181
   var array182: ptr ArrayT = func36(addr params0, getPath(), loopsFactor)
   debugReturn(array182.id)
   dealloc(params0.data)
   let loopLimit53 = (50) div (3) + 1
   for loop53 in 0 ..< loopLimit53:
      discard
      for i in 0 ..< array181.size:
         inc array181.data[i]
   for i in 0 ..< array181.size:
      if array181.data[i] == 95:
         return array181
   dec array181.refC
   if array181.refC == 0:
      dealloc(array181.data)
      dealloc(array181)
      debugFree(array181.id)
   return array182

